Pancakeswap-Agents/LargeLPTokenDepositWithdraw/src/agent.spec.ts

```diff
@@ -0,0 +1,540 @@
+import { FindingType, FindingSeverity, Finding, HandleTransaction } from "forta-agent";
+import { TestTransactionEvent } from "forta-agent-tools/lib/tests.utils";
+import { MockEthersProvider, createAddress } from "forta-agent-tools/lib/tests";
+
+import { BigNumber, ethers } from "ethers";
+import { provideHandleTransaction } from "./agent";
+import { MASTERCHEF_ABI, MASTERCHEF_ADDRESS, IBEP20_ABI } from "./constants";
+import { BotConfig } from "./config";
+
+const MASTERCHEF_INTERFACE = new ethers.utils.Interface(MASTERCHEF_ABI);
+const IBEP20_INTERFACE = new ethers.utils.Interface(IBEP20_ABI);
+
+// Function to set the token address of a certain token in the masterchef account
+function addLPTokenAddress(mockProvider: MockEthersProvider, pid: number, tokenAddress: string, block: number) {
+  mockProvider.addCallTo(MASTERCHEF_ADDRESS, block, MASTERCHEF_INTERFACE, "lpToken", {
+    inputs: [pid],
+    outputs: [tokenAddress],
+  });
+}
+
+// Function to set the name balance of masterchef account at a certain tokenAddress
+function addLPTokenNameBalance(
+  mockProvider: MockEthersProvider,
+  tokenAddress: string,
+  tokenName: string,
+  balance: ethers.BigNumber,
+  block: number
+) {
+  mockProvider.addCallTo(tokenAddress, block - 1, IBEP20_INTERFACE, "balanceOf", {
+    inputs: [MASTERCHEF_ADDRESS],
+    outputs: [balance],
+  });
+  mockProvider.addCallTo(tokenAddress, block, IBEP20_INTERFACE, "name", { inputs: [], outputs: [tokenName] });
+}
```

Comment:
You can separate these two calls, as in the Static mode you don't need to call `balanceOf`